### ISSUE TITLE
Resume the media download after cancelling it.

### ISSUE THREAD

**Issue by ashishbinu** [telegramdesktop/tdesktop#23977]
### Steps to reproduce

1. Download any big media file.
2. Cancel the download after the progress is considerably big.
3. Click download button again.

### Expected behaviour

It should start the download in step 3 from where it was stopped at step 2. This behaviour is there in Android Telegram App.

### Actual behaviour

When downloading the file again at step 3, it downloads the file from the beginning. All the old progress of download is lost.

### Operating system

Manjaro Linux x86_64, 21.2.1, bspwm

### Version of Telegram Desktop

3.4.0-1

### Installation source

Other (unofficial) source

**Comment by john-preston** [comment]
@ashishbinu Well, on the bright side the HDD space is freed if you cancel undesired download :)

**Comment by ashishbinu** [comment]
@john-preston By default, old cache is deleted as far as I know.

**Comment by john-preston** [comment]
@ashishbinu Files are downloaded not to cache, but to users Downloads folder. Nothing is automatically deleted from there :(

This is in case of Telegram Desktop. I don't know what about Android though.

**Comment by ashishbinu** [comment]
I am also not sure how the Android Telegram handles the partial file downloaded. But keeping this partial file for resuming later could be useful as I sometimes keep the big media files downloading and accidently close the telegram later not realising the the file was downloading and have to redownload the whole file again. If the partial file was there it could have downloaded from the previous progress and save broadband bandwidth.

**Comment by john-preston** [comment]
@ashishbinu Yes, it would be useful indeed. This issue may be kept as a feature request.

**Comment by ashishbinu** [comment]
@john-preston As far as I know, the mac client for telegram keeps the partial file in cache and puts it back to Downloads folder.

**Comment by ashishbinu** [comment]
> @john-preston As far as I know, the mac client for telegram keeps the partial file in cache and puts it back to Downloads folder after completing.

**Comment by john-preston** [comment]
@ashishbinu I like the idea apart from a case when the Downloads folder and users cache folder are on different volumes and the file won't be moved after download but would be copied instead.

**Comment by ashishbinu** [comment]
> @ashishbinu I like the idea apart from a case when the Downloads folder and users cache folder are on different volumes and the file won't be moved after download but would be copied instead.

Ohh yeah, that seems like a problem.

**Comment by M9Prod** [comment]
I really need a solution to avoid this issue.

**Comment by LordRusk** [comment]
the solution could be to be able to pick the finished "downloads" directory and cached directory before beginning the download...Probably in the pop up where you pick what files to download, and from when...

**Comment by fasakinhenry** [comment]
I still have this issue. A better way to handle this is to add an id for every download instance and when download is resumed it checks the download folder with that ID and resumes download and ensures keeping the files
